app/scripts/metamask-controller.js

```diff
@@ -2438,6 +2438,7 @@ export default class MetamaskController extends EventEmitter {
         allowedEvents: [],
       }),
       disabled: !this.preferencesController.state.useExternalServices,
+      getMetaMetricsId: () => this.metaMetricsController.getMetaMetricsId(),
```

Comment:
One side effect of this initialization is, `metametricsId` would be defined after we initialized the controller.